src/EnemyBehaviourEC.cpp

```diff
@@ -107,6 +155,34 @@ bool EnemyBehaviourEC::timeToAttack() {
     return false;
 }
 
+Ogre::Vector3 EnemyBehaviourEC::separate() {
+
+    Ogre::Vector3 resultado = Ogre::Vector3(0, 0, 0);
+    int numAgentes = otherTransform.size();
+    for (int i = 0; i < numAgentes; i++) {
+        TransformComponent* agenteObjetivo = otherTransform[i];
+
+        Ogre::Vector3 direccion =
+            dynamic_cast<TransformComponent*>(
+                this->father->getComponent("TransformComponent"))
+                ->getPosition() -
+            agenteObjetivo->getPosition();
```

Comment:
Split all of this code in multiple variables, long lines of code are wrapped like this, and in both ways it's hard to read.